source/developers-guide/plugin-system/index.md

```diff
@@ -205,6 +205,63 @@ class SwagSloganOfTheDay extends \Shopware\Components\Plugin
 ```
 
 
+### Decorate a service
+The following example shows how you can decorate a service with implements an interface and defined in the shopware di container. 
+```
+<?php
+
+namespace SwagExample\Bundle\StoreFrontBundle;
+
+use Shopware\Bundle\StoreFrontBundle\Service\ListProductServiceInterface;
+use Shopware\Bundle\StoreFrontBundle\Struct\ProductContextInterface;
+
+class ListProductService implements ListProductServiceInterface
+{
+    private $service;
+
+    public function __construct(ListProductServiceInterface $service)
+    {
+        $this->service = $service;
+    }
+
+    public function getList(array $numbers, ProductContextInterface $context)
+    {
+        $products = $this->service->getList($numbers, $context);
+        //...
+        return $products;
+    }
+
+    public function get($number, ProductContextInterface $context)
+    {
+        return array_shift($this->getList([$number], $context));
+    }
+}
+```
+The original `\Shopware\Bundle\StoreFrontBundle\Service\Core\ListProductService` defined with the service id `shopware_storefront.list_product_service`. The following service definition decorates this service with the above service:
+
+```
+<?xml version="1.0" ?>
+
+<container xmlns="http://symfony.com/schema/dic/services"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+    <services>
+        <service id="swag_example.list_product_service"
+                 class="SwagExample\Bundle\StoreFrontBundle\ListProductService"
+                 decorates="shopware_storefront.list_product_service"
+                 public="false">
+
+            <argument type="service" id="swag_example.list_product_service.inner"/>
```

Comment:
Maybe you can add a short hint that they must use the `.inner` of the new defined service instead of the original. I've seen people that were trying to use `shopware_storefront.list_product_service.inner` instead of their own